net/types.go

```diff
@@ -12,17 +12,25 @@ type SwapState = common.SwapStateNet //nolint:revive
 
 //nolint:revive
 type (
-	MessageType     = byte
-	Message         = common.Message
-	QueryResponse   = message.QueryResponse
-	SendKeysMessage = message.SendKeysMessage
+	MessageType        = byte
+	Message            = common.Message
+	QueryResponse      = message.QueryResponse
+	SendKeysMessage    = message.SendKeysMessage
+	RelayClaimRequest  = message.RelayClaimRequest
+	RelayClaimResponse = message.RelayClaimResponse
 )
 
-// Handler handles swap initiation messages.
-// It is implemented by *xmrmaker.xmrmaker
-type Handler interface {
+// MakerHandler handles swap initiation messages and offer queries. It is
+// implemented by *xmrmaker.xmrmaker.
+type MakerHandler interface {
 	GetOffers() []*types.Offer
-	HandleInitiateMessage(msg *SendKeysMessage) (s SwapState, resp Message, err error)
+	HandleInitiateMessage(msg *SendKeysMessage) (SwapState, Message, error)
+}
+
+// TakerHandler handles relay claim requests. It is implemented by
+// *xmrtaker.xmrtaker.
+type TakerHandler interface {
+	HandleRelayClaimRequest(msg *RelayClaimRequest) (*RelayClaimResponse, error)
```

Comment:
why is the relayer part of the taker? it shouldn't be, as it's not actually related logically; it should be its own instance

Reply:
We could do that, but here's my reasoning and what I thought we would do in followup work that might be better done in the taker:

Bob should always go directly to Alice first for help with claiming and that, at least with our software, Alice always helps even if she doesn't want to help relay claims from other swaps. Also, even if we use a separate instance, the 2 instances would be sharing Alice's wallet and the ability to lock it while transactions are being created/mined.

Reply:
I think it would make more sense to have a `relayer.Instance` or such that also contains the reference to the eth wallet, and either the database or the swap manager to tell if the relay request is from a current swap or not. relaying is logically separate from the taker, and thus shouldn't live in the same package imo

Reply:
also, the ethereum wallet (extendedethclient) is currently in the backend, so it's already used by both maker and taker anyways.

Reply:
Let's push this work into the follow-up PR that will ask Alice to relay Bob's claim before reaching out to the open market of relayers. We'll be modifying this code anyways and it will be easier to view in a smaller PR.